src/app/client/src/app/modules/notes/components/note-form/note-form.component.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { ResourceService, ToasterService, SharedModule } from '@sunbird/shared';
+import { ActivatedRoute } from '@angular/router';
+import { NotesService } from '../../services';
+import { UserService, LearnerService, CoreModule } from '@sunbird/core';
+import { Observable } from 'rxjs/Observable';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { FormsModule } from '@angular/forms';
+import { response } from './note-form-component.spec.data';
+import { NoteFormComponent } from './note-form.component';
+
+describe('NoteFormComponent', () => {
+  let component: NoteFormComponent;
+  let fixture: ComponentFixture<NoteFormComponent>;
+  const fakeActivatedRoute = {
+    'params': Observable.from([{ 'courseId': 'do_212347136096788480178', 'contentId': 'do_2123229899264573441612' }])
+  };
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ HttpClientTestingModule, FormsModule, SharedModule, CoreModule],
+      declarations: [NoteFormComponent],
+      providers: [UserService, ResourceService, ToasterService, NotesService,
+         LearnerService,
+        { provide: ActivatedRoute, useValue: fakeActivatedRoute }
+      ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(NoteFormComponent);
+    component = fixture.componentInstance;
+    spyOn(component, 'ngAfterViewInit');
+    fixture.detectChanges();
+  });
+
+
+  it('Should subscribe to note service while creating a note', () => {
+    const notesService = TestBed.get(NotesService);
+    const userService = TestBed.get(UserService);
+    const learnerService = TestBed.get(LearnerService);
+    spyOn(component.createEventEmitter, 'emit');
```

Comment:
input for courseid contentid and contentDetails for update and mode should be mocked and same should be checked from event emmiter